buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/GenerateCoveragePercentsTask.groovy

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.firebase.gradle.plugins.measurement
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+
+
+public class GenerateCoveragePercentsTask extends DefaultTask {
+
+    @OutputFile
+    File reportFile
+
+    @Input
+    List<String> packages
+
+    XmlSlurper parser
+
+    GenerateCoveragePercentsTask() {
+        parser = new XmlSlurper()
+        parser.setFeature("http://apache.org/xml/features/disallow-doctype-decl", false)
+        parser.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false)
+    }
+
+    @TaskAction
+    def generate() {
+        def pullRequestNumber = project.properties["pull_request"] ?: 777
```

Comment:
Maybe throw an error if a pull_request isn't provided?

Reply:
There are cases where we won't get a pull request number, such as debugging test locally. Added a comment there.

Reply:
For APK size, we always require a number. It can be passed manually to Gradle, and the number can even be fake. I'd suggest we have the same behavior here.